crates/uv/src/commands/project/init.rs

```diff
@@ -0,0 +1,169 @@
+use std::fmt::Write;
+
+use anyhow::Result;
+
+use owo_colors::OwoColorize;
+use pep508_rs::PackageName;
+use uv_cache::Cache;
+use uv_client::{BaseClientBuilder, Connectivity};
+use uv_configuration::PreviewMode;
+use uv_distribution::pyproject_mut::PyProjectTomlMut;
+use uv_distribution::{ProjectWorkspace, WorkspaceError};
+use uv_fs::Simplified;
+use uv_python::{
+    EnvironmentPreference, PythonFetch, PythonInstallation, PythonPreference, PythonRequest,
+};
+use uv_warnings::warn_user_once;
+
+use crate::commands::ExitStatus;
+use crate::printer::Printer;
+
+/// Add one or more packages to the project requirements.
+#[allow(clippy::too_many_arguments, clippy::fn_params_excessive_bools, clippy::single_match_else)]
+pub(crate) async fn init(
+    path: Option<String>,
+    name: Option<PackageName>,
+    no_readme: bool,
+    no_pin: bool,
+    python: Option<String>,
+    python_preference: PythonPreference,
+    python_fetch: PythonFetch,
+    preview: PreviewMode,
+    connectivity: Connectivity,
+    native_tls: bool,
+    cache: &Cache,
+    printer: Printer,
+) -> Result<ExitStatus> {
+    if preview.is_disabled() {
+        warn_user_once!("`uv init` is experimental and may change without warning.");
+    }
+
+    // Discover the current workspace, if it exists.
+    let current_dir = std::env::current_dir()?.canonicalize()?;
+    let workspace = match ProjectWorkspace::discover(&current_dir, None).await {
+        Ok(project) => Some(project),
+        Err(WorkspaceError::MissingPyprojectToml) => None,
+        Err(err) => return Err(err.into()),
+    };
+
+    let canonical_path = match path {
+        None => current_dir.clone(),
+        Some(ref path) => current_dir.join(path),
+    };
+
+    let name = match name {
+        Some(name) => name,
+        None => {
+            // Get the name of the directory.
+            let name = canonical_path
+                .file_name()
+                .and_then(|path| path.to_str())
+                .expect("Invalid package name");
+            PackageName::new(name.to_string())?
+        }
+    };
+
+    // Make sure the package does not already exist.
+    if canonical_path.join("pyproject.toml").exists() {
+        anyhow::bail!("Package is already initialized")
+    }
+
+    // Create the directory for the project.
+    let src_dir = canonical_path.join("src").join(name.as_ref());
+    fs_err::create_dir_all(&src_dir)?;
+
+    // Create the `pyproject.toml`.
+    fs_err::write(
+        canonical_path.join("pyproject.toml"),
+        indoc::formatdoc! {r#"
+        [project]
+        name = "{name}"
+        version = "0.1.0"
+        description = "Add your description here"
+        dependencies = []
+        readme = "README.md"
```

Comment:
Anything else we should add here? I think Rye and Poetry also fill in the authors field, probably through git config?

Reply:
`uv init` also doesn't currently run `git init` or support `--vcs`, which cargo does.

Reply:
> Anything else we should add here? I think Rye and Poetry also fill in the authors field, probably through git config?

The command could prompt, using the git config or example, example@example.com as a default.

Reply:
I'm not sure if it's within the scope of this PR, but there could be a boolean prompt that could lead to further prompting to add package-specific fields such as keywords, classifiers, urls

Reply:
The other thing missing is `requires-python`, should we default to `>=3.12`?

Reply:
uv has a default python executable, perhaps requires-python could be set to that?

Reply:
Defaulting to the latest seems.. reasonable.